ROADMAP.md

```diff
@@ -0,0 +1,117 @@
+
+I will go into more detail but before that; a simplistic example. These would be the steps for wrapping method `x`:
+
+1. In the Swift SDK, add method `wrapsX`. It will wrap method `x` imported from the Objective-C SDK.
+2. In the Objective-C SDK add underscored prefix method `__x` that wraps method `x`.
+3. In the Objective-C SDK, tag method `x` with the `FB_SWIFT_DEPRECATED` macro shown below.
+    ```
+    #if __swift__
+	#define FB_SWIFT_DEPRECATED(message) __attribute__((deprecated(message)));
+	#else
+	#define FB_SWIFT_DEPRECATED(...)
+	#endif
+	```
+3. In the Swift SDK, update method `wrapsX` to call `__x`
+4. In the Objective-C SDK, replace the `FB_SWIFT_DEPRECATED` macro with the [`NS_REFINED_FOR_SWIFT`](https://developer.apple.com/documentation/swift/objective-c_and_c_code_customization/improving_objective-c_api_declarations_for_swift) macro.
+5. In the Objective-C SDK, delete method `__x` as it will be provided by `NS_REFINED_FOR_SWIFT`.
+
+If that all made perfect sense to you then feel free to skip the detailed example below.
+
+## Detailed Example
+
+Consider, for instance, a function in Objective-C that accepts a  pointer to a user.
+
+`-(void)setUser:(User *)user;` 
+
+This is inherently unsafe. Also, in our pretend example we'd like to use a `ValidatedUser` struct. How can we update our API to encourage developers to use a `ValidatedUser` instead of a pointer?
+
+First we go into the Objective-C SDK and add the `FB_SWIFT_DEPRECATED` macro.
+
+`-(void)setUser:(User *)user FB_SWIFT_DEPRECATED("This will be replaced by 'setValidatedUser(_:)'");`
+
+Now, if you try and use this method from Swift, you will get a warning. Good.
+
+We then wrap the original call like so:
+
+```
+/// Artificially 'refined'. ie. can be used from Swift prior to using `NS_REFINED_FOR_SWIFT` macro
+- (void)__-(void)setUser:(User *)user {
+    [self setUser:user];
+}
+```
+
+With this in place we can move our efforts to the Swift project and wrap the artificially refined version of the method:
+
+```
+func setValidated(_ validUser: ValidatedUser) -> Void {
+	// Any number of strategies can be employed here to implement a layer of validation 
+	// and type-safety. 
+	// This assumes that we can simply use data from our validated user to construct
+	// an instance of the Objective-C User class.
+ 	let user = User(name: validUser.name, age: validUser.age)
+   
+   // Utilizes the original Objective-C implementation. 
+   // The __ is a naming convention associated with NS_REFINED_FOR_SWIFT
+   __setUser(&user) 
+}
+```
+
+Now this is better but not perfect. We are providing Swift developers with a safer way to set a `User` but we also expose the unsafe way.
+
+We can fix this by going back into the Objective-C project and changing `FB_SWIFT_DEPRECATED` to `NS_REFINED_FOR_SWIFT`. The compiler will force us to delete our underscore-prefixed wrapping method.
```

Comment:
```suggestion
We can fix this by going back into the Objective-C project and changing `FB_SWIFT_DEPRECATED` to `NS_REFINED_FOR_SWIFT`. The compiler will force us to delete our temporary underscore-prefixed wrapping method, since `NS_REFINED_FOR_SWIFT` will automatically expose the original `setUser:` method as `__setUser(_:)` to Swift.
```

Reply:
👏👏👏👏